Source/DafnyRuntime/DafnyRuntime.py

```diff
@@ -165,8 +191,8 @@ def __contains__(self, item):
         return self[item] > 0
 
 class Map(dict):
-    def __str__(self) -> str:
-        return 'map[' + ', '.join(map(lambda i: f'{i[0]} := {i[1]}', self.items)) + ']'
+    def __dafnystr__(self) -> builtins.str:
+        return 'map[' + ', '.join(map(lambda i: f'{str(i[0])} := {str(i[1])}', self.items)) + ']'
```

Comment:
I find `str` risky as a name choice.  Are we sure we will remember to use `builtins.str` everywhere in the prelude?  I think it would be safer to use `dstr()` or `to_str`, and leave `str` to have its usual meaning?